### ISSUE TITLE
predict time 

### ISSUE THREAD

**Issue by 123liluky** [matterport/Mask_RCNN#670]
I use code in model.py to test predict time for an image like below:
t1 = time.time()
detections, mrcnn_class, mrcnn_bbox, mrcnn_mask, \
            rois, rpn_class, rpn_bbox =\
            self.keras_model.predict([molded_images, image_metas], verbose=0)
print('predict_time:{:.3f}s'.format(time.time()-t1))

The results are time for the first image is 3.3s and time for images from the second is all around 0.065s. I am wondering why time for the first image is much larger? Thank you.

**Comment by wenston2006** [comment]
I came across with similar problem recently. I downgraded the keras2.2.2 to keras2.0.8, then the code could run successfully. But I ignored the inconformity between Keras application and Keras.

**Comment by zungam** [comment]
Most likly more masks. How many masks did they find?

**Comment by 123liluky** [comment]
There is one mask in each image. Are there any other possible reasons?

**Comment by zungam** [comment]
Are the masks same size? In pixels?
Are the images same size?

**Comment by 123liluky** [comment]
Yes, the first and second images and masks from are very close. Have I add time predicting code wrongly?

**Comment by xiaolisarah** [comment]
i want ask you predict time 0.065s in what kind of GPU  ＠123liluky

**Comment by 123liluky** [comment]
@xiaolisarah 2G GTX1050

**Comment by xiaolisarah** [comment]
wooo that's sounds great,i have poor predict time,can u show you predict code

**Comment by xiaolisarah** [comment]
@123liluky wooo that's sounds great,i have poor predict time,can u show you predict code

**Comment by xiaolisarah** [comment]
@123liluky what's you detect time per image ?

**Comment by 123liluky** [comment]
Nearly 100ms

**Comment by xiaolisarah** [comment]
@123liluky my predict time nearly 800ms,that's awf

**Comment by xiaolisarah** [comment]
@123liluky what's you image size

**Comment by 123liluky** [comment]
256*256*3

**Comment by mehditlili** [comment]
That is normal, keras needs to allocate memory and do other stuff before the first inference, check [this issue](https://github.com/keras-team/keras/issues/8724)
